### PR TITLE
Changes to create diagram with graphviz Addresses: #164

### PR DESCRIPTION
**Pull request by GadgetSteve**
Changes to conf.py to:

  1. Enable graphviz extension 
  2. Set graphviz output to SVG format
  3. Add some comments on how to get graphviz working from sphinx on windows

Changes to docs/source/projects/content-projects.rst to replace current, (neater), png with graphviz section that produces .svn image which while not as neat does have working links.

Tested for: html, htmldir, htmlsingle & epub on Windows 10.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -62,7 +63,7 @@
 # The short X.Y version.
 version = '4.1'
 # The full version, including alpha/beta/rc tags.
-release = '4.1.0'
```
Comment (Carreau) [inline docs/source/conf.py]: We would need to change that I think.

**Comment by Carreau** [comment]
Hi @GadgetSteve thanks a lot for doing that !

Is graphviz available on read the docs ? Otherwise that would break the autobuild. 

@willingc you are the documentation master, if this is fine with you feel free to go ahead with it .

**Comment by GadgetSteve** [comment]
@Carreau - A quick check on [Read The Docs](http://read-the-docs.readthedocs.io/en/latest/builds.html#packages-installed-in-the-build-environment) lists graphviz as one of the installed libraries so it **_should_** just "work out of the box".

File: docs/source/projects/content-projects.rst [hunk 2]
```diff
@@ -13,8 +13,54 @@ Installation
...
-   :width: 100%
+How do I decide which packages I need?
+--------------------------------------
+
+.. graphviz::
+
+	digraph decide_project {
+		size="10,4";
+		// Top Level
+		//label="How do I decide which pakages I need?";
```
Comment (willingc) [inline docs/source/projects/content-projects.rst]: @GadgetSteve Typo in spelling of `packages` in this line and the next line.

File: docs/source/projects/content-projects.rst [hunk 3]
```diff
@@ -13,8 +13,54 @@ Installation
...
+		try [shape=box, label="Try the notebook"]
+		install [shape=box, label="Install the notebook", tooltip="Go To Install Jupyter"]
+		team [shape=box, label="Install notebooks\n for my team, class,\n or group"]
+		convert [shape=box, label="Convert my\n notebook file to\n another format"]
+		lang [shape=box, label="Use another\n programming\n language such as\n R or Java"]
+		custom [shape=box, label="Customize the\n notebook for my\n needs"]
+		// 3rd Level
+		online [label="Try Online", href="https://try.jupyter.org/", target="_blank", tooltip="Try Online", style=underline, color=blue,]
+		doinstall [label="Install Jupyter", tooltip="How to install", href="../install.html", target="_top", color=green,] // 4th level
+		hub [label="Jupyter Hub", tooltip="Install Jupyter Hub", target="_blank", color=blue, href="https://github.com/jupyterhub/jupyterhub"]
```
Comment (willingc) [inline docs/source/projects/content-projects.rst]: JupyterHub should be one word.

File: docs/source/projects/content-projects.rst [hunk 4]
```diff
@@ -13,8 +13,54 @@ Installation
...
+		nbconvert [label="nbconvert", tooltip="How to convert notebooks", target="_blank", color=blue, href="https://nbconvert.readthedocs.io/en/latest/"]
+		kernel [label="Install a\n language\nkernel", tooltip="How to install kernels", target="_top", color=green, href="../projects/subprojects.html?highlight=jupyterhub#kernels"]
+		noinst [shape=plaintext, label="No installation\nneeded!", color=blue]
+		like [shape=plaintext, label="I like it,\nand I wish to install it!", color=blue]
+		my [shape=plaintext, label="On my system", tooltip="Local Installation", color=blue]
+		// below kernel
+		widgets [label="Widgets", tooltip="Install & use ipywidgets", target="_blank", color=blue, href="https://ipywidgets.readthedocs.io/en/latest/"]
+		extend [label="Extensions", tooltip="Install & use extentions", target="_blank", color=blue, href="https://github.com/jupyter/help"]
```
Comment (willingc) [inline docs/source/projects/content-projects.rst]: Typo in tooltip: should be `extensions`

**Comment by willingc** [comment]
@GadgetSteve I'm sorry for the delay in reviewing and thank you for the contribution :smile:  Thanks @Carreau for doing the initial review and following up.

I've left a few comments on the code (mostly typos). I am currently traveling, but in the next couple days, I will check the `href` links a bit more closely and compare them to the links in the original visual.

I appreciate your PR and interest in improving our documentation. I'm looking forward to getting this merged after correcting the typos and links.

**Comment by GadgetSteve** [comment]
@willingc & @Carreau - thank you for the kind words:

 - I have corrected the typos noted above _spelling was never my strong point_ 
 - Laid things out in a more readable, pythonic and consistent manner.
 - Added a key to the colours of the link shapes.
 - Added a font selection which I think makes it much more readable.
 - Added missing tool tips to several of the items.

**Comment by GadgetSteve** [comment]
Please note that the Travis build system does show one warning due to being unable to find 'dot' on the travis build system - I would guess that adding to .travis.yml, near the beginning, something like:

before_install:
	- sudo apt-get update -qq
	- sudo apt-get install graphviz 

Should do the job but I have very limited experience of travis-ci so have no way of knowing if this is a container based build _if it is then the command syntax will be different - see [here](https://docs.travis-ci.com/user/installing-dependencies/#Installing-Packages-on-Container-Based-Infrastructure)_

**Comment by willingc** [comment]
@GadgetSteve I'm happy to merge this PR. Thanks for improving the docs and creating active links. I hope you consider contributing again in the future. :cake: 🎉

**Comment by GadgetSteve** [comment]
My pleasure Carol _@willingc_ it also provided me with a useful learning experience of both the dot language and of the sphinx build system. I also found out about some Jupyter features that I was unaware of in the process which is always to the good. Thank-you for your awesome work on the Jupyter project along with all of the devs.

**Pull request merged**